src/toil/batchSystems/lsf.py

```diff
@@ -24,274 +24,171 @@
-            'rusage[mem=' + str(mem_resource) + ']" -M' + str(mem_limit)
-    else:
-        bsubMem = ''
-    cpuStr = '' if cpu is None else '-n ' + str(int(cpu))
-    bsubline = ["bsub", bsubMem, cpuStr,"-cwd", ".", "-o", "/dev/null", "-e",
-        "/dev/null"]
-    lsfArgs = os.getenv('TOIL_LSF_ARGS')
-    if lsfArgs:
-        bsubline.extend(lsfArgs.split())
-    return bsubline
-
-
-def bsub(bsubline):
-    process = subprocess.Popen(" ".join(bsubline), shell=True, stdout = subprocess.PIPE, stderr = subprocess.STDOUT)
-    liney = process.stdout.readline()
-    logger.debug("BSUB: " + liney)
-    result = int(liney.strip().split()[1].strip('<>'))
-    logger.debug("Got the job id: %s" % (str(result)))
-    return result
-
-
-def getjobexitcode(lsfJobID):
-        job, task = lsfJobID
-
-        # first try bjobs to find out job state
-        args = ["bjobs", "-l", str(job)]
-        logger.debug("Checking job exit code for job via bjobs: " + str(job))
-        process = subprocess.Popen(" ".join(args), shell=True, stdout = subprocess.PIPE, stderr = subprocess.STDOUT)
-        started = 0
-        for line in process.stdout:
-            if line.find("Done successfully") > -1:
-                logger.debug("bjobs detected job completed for job: " + str(job))
-                return 0
-            elif line.find("Completed <exit>") > -1:
-                logger.debug("bjobs detected job failed for job: " + str(job))
-                return 1
-            elif line.find("New job is waiting for scheduling") > -1:
-                logger.debug("bjobs detected job pending scheduling for job: " + str(job))
-                return None
-            elif line.find("PENDING REASONS") > -1:
-                logger.debug("bjobs detected job pending for job: " + str(job))
+from toil.batchSystems.abstractGridEngineBatchSystem import \
+        AbstractGridEngineBatchSystem
+from toil.batchSystems.lsfHelper import (parse_memory_resource,
+                                         parse_memory_limit,
+                                         per_core_reservation)
+
+logger = logging.getLogger(__name__)
+
+
+class LSFBatchSystem(AbstractGridEngineBatchSystem):
+
+    class Worker(AbstractGridEngineBatchSystem.Worker):
+        """LSF specific AbstractGridEngineWorker methods."""
+
+        def getRunningJobIDs(self):
+            times = {}
+            currentjobs = dict((str(self.batchJobIDs[x][0]), x) for x in
+                               self.runningJobs)
+            process = subprocess.Popen(
+                    ["bjobs", "-o", "jobid stat start_time delimiter='|'"],
+                    stdout=subprocess.PIPE)
+            stdout, _ = process.communicate()
+
+            for curline in stdout.split('\n'):
+                items = curline.strip().split('|')
+                if items[0] in currentjobs and items[1] == 'RUN':
+                    jobstart = parse(items[2], default=datetime.now(tzlocal()))
```

Comment:
`dateutil.parser.parse` is really lovely and perfect. I'll make a PR for the other GridEngine-esque batchsystems to make use of it after this is merged.